src/admin-user-roles/admin-user-roles.routes.js

```diff
@@ -0,0 +1,118 @@
+                    return new ObjectMapper().map(roles, 'rights');
+                }
+            },
+            views: {
+                '@openlmis': {
+                    templateUrl: 'admin-user-roles/user-roles.html',
+                    controller: 'UserRolesController',
+                    controllerAs: 'vm'
+                }
+            }
+        });
+        // MALAWISUP-3888 Add checkboxes and column filters under roles when removing users roles: Starts here
+        addStateForRoleType(ROLE_TYPES.SUPERVISION, '/supervision?page&size&programId&roleId&storageKey', 'user-roles-supervision.html');
+        addStateForRoleType(ROLE_TYPES.ORDER_FULFILLMENT, '/fulfillment?page&size&programId&roleId&storageKey', 'user-roles-fulfillment.html');
+        addStateForRoleType(ROLE_TYPES.GENERAL_ADMIN, '/admin?page&size&programId&roleId&storageKey', 'user-roles-tab.html');
+        addStateForRoleType(ROLE_TYPES.REPORTS, '/reports?page&size&programId&roleId&storageKey', 'user-roles-tab.html');
+        function addStateForRoleType(type, url, templateFile) {
+            $stateProvider.state('openlmis.administration.users.roles.' + type, {
+                label: ROLE_TYPES.getLabel(type),
+                url: url,
+                controller: 'UserRolesTabController',
+                templateUrl: 'admin-user-roles/' + templateFile,
+                controllerAs: 'vm',
+                resolve: {
+                    tab: function() {
+                        return type;
+                    },
+                    filteredRoleAssignments: function($stateParams, user, tab) {
+                        var assignments = user.getRoleAssignments(tab);
+
+                        if ($stateParams.programId) {
+                            assignments = assignments.filter(function(assignment) {
+                                return assignment.programId === $stateParams.programId;
+                            });
+                        }
+
+                        if ($stateParams.roleId) {
+                            console.log(assignments[0])
+                            assignments = assignments.filter(function(assignment) {
+                                return assignment.roleId === $stateParams.roleId;
+                            });
+                        }
+
+                        return assignments;
+                    },
+                    roleAssignments: function(paginationService, $stateParams, filteredRoleAssignments) {
+                        return paginationService.registerList(null, $stateParams, function() {
+                            return filteredRoleAssignments;
+                        });
+                    },
+                    filteredRoles: function(roles, tab) {
+                        return roles.filter(function(role) {
+                            return role.type === tab;
+                        });
+                    }
+                }
+                // MALAWISUP-3888: Ends here
```

Comment:
Could you move this comment to the place when the customization ends and possibly add a new bloc of comments where there are other modifications. I believe that between staring comment and ending there is not modified code. Please change it in other places as well